src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -282,6 +290,97 @@ private IEnumerable<BlittableJsonReaderObject> ReadArray(INewDocumentActions act
             }
         }
 
+        private IEnumerable<DocumentItem> ReadLegacyAttachments(INewDocumentActions actions)
+        {
+            if (UnmanagedJsonParserHelper.Read(_stream, _parser, _state, _buffer) == false)
+                ThrowInvalidJson("Unexpected end of json");
+
+            if (_state.CurrentTokenType != JsonParserToken.StartArray)
+                ThrowInvalidJson("Expected start array, but got " + _state.CurrentTokenType);
+
+            var context = _context;
+            var modifier = new BlittableMetadataModifier(context);
+            var builder = CreateBuilder(context, modifier);
+            try
+            {
+                while (true)
+                {
+                    if (UnmanagedJsonParserHelper.Read(_stream, _parser, _state, _buffer) == false)
+                        ThrowInvalidJson("Unexpected end of json while reading legacy attachments");
+
+                    if (_state.CurrentTokenType == JsonParserToken.EndArray)
+                        break;
+
+                    if (actions != null)
+                    {
+                        var oldContext = context;
+                        context = actions.GetContextForNewDocument();
+                        if (oldContext != context)
+                        {
+                            builder.Dispose();
+                            builder = CreateBuilder(context, modifier);
+                        }
+                    }
+                    builder.Renew("import/object", BlittableJsonDocumentBuilder.UsageMode.ToDisk);
+
+                    ReadObject(builder);
+
+                    var data = builder.CreateReader();
+                    builder.Reset();
+
+                    var attachment = new DocumentItem.AttachmentStream
+                    {
+                        Stream = actions.GetTempStream()
+                    };
+
+                    var attachmentInfo = ProcessLegacyAttachment(context, data, ref attachment);
+                    var dummyDoc = new DocumentItem()
+                    {
+                        Document = new Document()
+                        {
+                            Data = WriteDummyDocumentForAttachment(context, attachmentInfo),//TODO:this is wrong i need to generate a dummy document just like in the DR tool
```

Comment:
what is the meaning here?

Reply:
I forgot to remove the comment, this is already the fix.